odoo/fields.py

```diff
@@ -1760,11 +1848,20 @@ def _convert_from_cache_to_column(self, value):
     def convert_to_cache(self, value, record, validate=True):
         if value is None or value is False:
             return None
+        if self.translate:
+            lang = (record.env.lang or 'en_US') if self.translate is True else record.env._lang
+            return {lang: value}
         return value
 
     def convert_to_record(self, value, record):
         if value is None:
             return False
+        if self.translate:
+            lang = (record.env.lang or 'en_US') if self.translate is True else record.env._lang
+            try:
+                value = value[lang]
+            except KeyError:
+                raise CacheMiss(record._name, record._ids, self) from None
```

Comment:
CacheMiss will be rasied in the converter for cache value
`convert_to_record(convert_to_cache(value))` should never have CacheMiss